src/dune_lang/package_name.ml

```diff
@@ -49,11 +49,24 @@ module Opam_compatible = struct
 
     let of_string_opt s = Option.some_if (is_valid_string s) s
 
-    let make_valid s =
+    let escape s =
       let replaced = String.map s ~f:(fun c -> if is_valid_char c then c else '_') in
       if is_valid_string replaced then replaced else "p" ^ replaced
     ;;
 
+    let make_valid s =
+      match index_opt s '.' with
```

Comment:
Better use `String.lsplit2 ~on:'.' s` here, as this will get you the `before`/`after` part immediately.

Reply:
Didn't know this one, as it's in stdune but not in the stdlib :)